File: Java/Language/Inheritance/AbstractClassTest/src/CurrentAccount.java
package edu.met.banking;

//a class declared with final modifier cannot be sub-class
final class CurrentAccount extends Account {

	public void deposit(double amount) {
		if(balance < 0)
			amount *= 0.95;
		balance += amount;
	}

	public void withdraw(double amount) {
		balance -= amount;
	}
}

File: dotnet/Language/Inheritance/AbstractClassTest/BankLib/CurrentAccount.cs
namespace Met.Banking;

//a class defined with sealed modifier cannot be
//used as a base class
sealed class CurrentAccount : Account
{
    public override void Deposit(double amount)
    {
        Balance += amount;
    }

    public override void Withdraw(double amount)
    {
        Balance -= amount;
    }
}